### DISCUSSION TITLE
Aspera Connect クライアント(ascp)のスパコン上からの利用

### DISCUSSION THREAD

**Discussion by sc-fs** [nig-sc/nigsc_homepage2#290] [Q&A] [answered]
(sc-helpdeskへの質問を質問者の許可を得て代理で起票しております)

スパコン上でaspera connectを使用したく下記を参照にしましたが，エラーが出てしまいました．
参考にしたもの：https://sc.ddbj.nig.ac.jp/software/aspera/install_Aspera#%E3%82%AF%E3%83%A9%E3%82%A4%E3%82%A2%E3%83%B3%E3%83%88%E8%A8%88%E7%AE%97%E6%A9%9F%E3%81%8C-linux-%E3%81%AE%E5%A0%B4%E5%90%88

インストールし，パスを通した過程です．
curl -O https://d3gcli72yxqn2z.cloudfront.net/downloads/connect/latest/bin/ibm-aspera-connect_4.2.4.265_linux.tar.gz
tar zxvf ibm-aspera-connect_4.2.4.265_linux.tar.gz
bash ibm-aspera-connect_4.2.4.265_linux.tar.sh
ここで$HOME/.aspera/connect/binが生成されていました．
しかし，下記のようになってしまいました．
$ ascp --help
ascp: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by ascp)
ascp: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by ascp)

スパコン上でも Aspera connect の Web ブラウザプラグインが必要ということでしょうか？
その場合，このやり方を存じ上げておらず，ご教示いただければ幸いです．
よろしくお願い致します．

**Comment by oogasawa** [comment, marked as answer]
Aspera Connect version 4の現在の動作環境がCentOS 8以降となっていることが原因です。
Release Notes: IBM Aspera Connect 4.2.4
https://www.ibm.com/docs/en/aspera-connect/4.2?topic=release-notes-aspera-connect-424
![Screenshot 2023-02-18 15:43:45](https://user-images.githubusercontent.com/35567475/219845778-7544beee-46a7-4f88-93a2-5b87509d70db.png)


遺伝研スパコンはCentOS 7で動いているので、このままではascpを動作させることができません。

---

ascpのApptainer (旧Singularity)コンテナイメージを作ってみました。これを遺伝研スパコン上に設置することでascpを使うことができます。
- https://github.com/nig-sc/docker_ascp4

以下のようにしてコンテナイメージを遺伝研スパコンのホームディレクトリ中の適当な場所に置きます。

```
cd $HOME
wget  https://github.com/nig-sc/docker_ascp4/raw/main/ascp4_ubuntu22.sif
```

Aspera Connectのインストーラはユーザ権限で実行する必要があり、`$HOME/.aspera`の下にインストールされるので、
最初の一回はインストール作業が必要なようです。以下のコマンドにより`ascp`コマンドがインストールされます。

```
$ apptainer exec ascp4_ubuntu22.sif bash /usr/local/src/ibm-aspera-connect_4.2.4.265_linux.sh

Installing IBM Aspera Connect

Deploying IBM Aspera Connect (/home/you/.aspera/connect) for the current user only.
Unable to register protocol handler, IBM Aspera Connect won't be able to auto-launch
Unable to register protocol handler, IBM Aspera Connect won't be able to support Faspex 5 transfers
Unable to update desktop database, IBM Aspera Connect may not be able to auto-launch

Install complete.
```

以下のようにすると`ascp`コマンドを実行することができます。

```
$ apptainer exec ascp4_ubuntu22.sif  $HOME/.aspera/connect/bin/ascp --help
Usage: ascp [OPTION] SRC... DEST
          SRC to DEST, or multiple SRC to DEST dir
          SRC, DEST format: [[user@]host:]PATH
  -h,--help                       Display usage
  -A,--version                    Display version.
  -T                              Disable encryption
  -d                              Create target directory, implied for file/file-pair lists
  -p                              Preserve file timestamp
  -q                              Disable progress display
  -v                              Verbose mode
(以下略)
```

**Reply by oogasawa** [reply]
apptainerコマンドが見つからない場合はmodule loadを使って明示的にロードしてください。

```
$ module load apptainer/1.1.0
```
